_posts/news/2023-09-30-announcing-scalajs.1.14.0.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title: Announcing Scala.js 1.14.0
+category: news
+tags: [releases]
+permalink: /news/2023/09/30/announcing-scalajs-1.14.0/
+---
+
+We are pleased to announce the release of Scala.js 1.14.0!
+
+This release mainly brings speed improvements in the reachability analysis by parallizing it.
+TODO Mention 2.13.12 upgrade.
+
+Read on for more details.
+
+<!--more-->
+
+## Getting started
+
+If you are new to Scala.js, head over to [the tutorial]({{ BASE_PATH }}/tutorial/).
+
+If you need help with anything related to Scala.js, you may find our community [in `#scala-js` on Discord](https://discord.com/invite/scala) and [on Stack Overflow](https://stackoverflow.com/questions/tagged/scala.js).
+
+Bug reports can be filed [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Release notes
+
+If upgrading from Scala.js 0.6.x, make sure to read [the release notes of Scala.js 1.0.0]({{ BASE_PATH }}/news/2020/02/25/announcing-scalajs-1.0.0/) first, as they contain a host of important information, including breaking changes.
+
+This is a **minor** release:
+
+* It is backward binary compatible with all earlier versions in the 1.x series: libraries compiled with 1.0.x through 1.13.x can be used with 1.14.0 without change.
+* Despite being a minor release, 1.14.0 is forward binary compatible with 1.13.x. It is *not* forward binary compatible with 1.12.x. Libraries compiled with 1.14.0 can be used with 1.13.x but not with 1.12.x or earlier.
+* It is *not* entirely backward source compatible: it is not guaranteed that a codebase will compile *as is* when upgrading from 1.13.x (in particular in the presence of `-Xfatal-warnings`).
+
+As a reminder, libraries compiled with 0.6.x cannot be used with Scala.js 1.x; they must be republished with 1.x first.
+
+## Improvements
+
+### Reachability Analysis is parallel by default
+
+We have observed up to 2x speedup of reachability analysis on the Scala.js test suite with this change.
```

Comment:
I'm wondering if the 2x is too marketing-y here. It is true, but most people will not care about the reachability analysis in isolation but about the whole linking process.

Reply:
Maybe mention how much of the whole linker is taken up by the analysis?

Reply:
Yes :-/ Do we have these numbers?

Reply:
Seems we do: https://github.com/scala-js/scala-js/pull/4897#issuecomment-1685227514 :) I'll run some analysis.

Reply:
```
> d %>%
  filter(!grepl(':', op)) %>%
  group_by(variant, op) %>%
  summarise(t_ns = median(t_ns)) %>%
  group_by(variant) %>%
  summarise(t_ns = sum(t_ns))
  
`summarise()` has grouped output by 'variant'. You can override using the `.groups` argument.
# A tibble: 3 × 2
  variant        t_ns
  <fct>         <dbl>
1 main     1090421195
2 parallel  835999739
3 pr        866628092
```

So overall a 20% speedup.

Reply:
```suggestion
We have observed up to 2x speedup of reachability analysis on the Scala.js test suite with this change, which corresponds to a 20% speedup to the overall linker.
```